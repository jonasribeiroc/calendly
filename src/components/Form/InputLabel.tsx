import React from 'react';
import styled from 'styled-components';

interface InputLabelProps {
    htmlFor: string;
    label: string;
    required?: boolean;
}

const StyledLabel = styled.label`
    display: block;
    margin-bottom: 0.5rem;
    font-weight: 600;
    color: #333;
`;

const RequiredAsterisk = styled.span`
    color: #333;
    margin-left: 0.25rem;
`;

const InputLabel: React.FC<InputLabelProps> = ({ htmlFor, label, required = false }) => {
    return (
        <StyledLabel htmlFor={htmlFor}>
            {label}
            {required && <RequiredAsterisk>*</RequiredAsterisk>}
        </StyledLabel>
    );
};

export default InputLabel;
