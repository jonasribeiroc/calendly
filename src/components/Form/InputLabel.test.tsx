import React from 'react';
import { render, screen } from '@testing-library/react';
import { InputLabel } from './InputLabel';

describe('InputLabel Component', () => {
  it('renders the label with correct text', () => {
    render(<InputLabel htmlFor="test-input" label="Test Label" />);
    expect(screen.getByText('Test Label')).toBeInTheDocument();
  });

  it('associates the label with the correct htmlFor', () => {
    render(<InputLabel htmlFor="test-input" label="Test Label" />);
    const label = screen.getByText('Test Label');
    expect(label).toHaveAttribute('for', 'test-input');
  });

  it('shows asterisk when required is true', () => {
    render(<InputLabel htmlFor="test-input" label="Test Label" required />);
    expect(screen.getByText('*')).toBeInTheDocument();
  });

  it('does not show asterisk when required is false', () => {
    render(<InputLabel htmlFor="test-input" label="Test Label" />);
    expect(screen.queryByText('*')).not.toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<InputLabel htmlFor="test-input" label="Test Label" required />);
    expect(asFragment()).toMatchSnapshot();
  });
});
