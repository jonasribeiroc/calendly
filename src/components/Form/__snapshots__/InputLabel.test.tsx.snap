// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel Component matches the snapshot 1`] = `
<DocumentFragment>
  <label
    class="sc-blHHSb xrmhh"
    for="test-input"
  >
    Test Label
    <span
      class="sc-gtLWhw ezURQU"
    >
      *
    </span>
  </label>
</DocumentFragment>
`;
